packages/components/src/components/form/bal-number-input/bal-number-input.tsx

```diff
@@ -0,0 +1,338 @@
+   */
+  @Prop() debounce = 0
+
+  @Watch('debounce')
+  protected debounceChanged() {
+    this.balChange = debounceEvent(this.balChange, this.debounce)
+  }
+
+  /**
+   * The value of the input.
+   */
+  @Prop({ mutable: true }) value?: number | null = undefined
+
+  /**
+   * Update the native input element when the value changes
+   */
+  @Watch('value')
+  protected valueChanged(newValue: number | null | undefined, oldValue: number | null | undefined) {
+    if (this.didInit && newValue !== oldValue && !this.hasFocus) {
+      this.balChange.emit(this.value)
+    }
+  }
+
+  /**
+   * Emitted when a keyboard input occurred.
+   */
+  @Event() balInput!: EventEmitter<number | null | undefined>
+
+  /**
+   * Emitted when the value has changed.
+   */
+  @Event() balChange!: EventEmitter<number | null | undefined>
+
+  /**
+   * Emitted when the input loses focus.
+   */
+  @Event() balBlur!: EventEmitter<FocusEvent>
+
+  /**
+   * Emitted when the input has focus.
+   */
+  @Event() balFocus!: EventEmitter<FocusEvent>
+
+  /**
+   * Emitted when a keyboard key has pressed.
+   */
+  @Event() balKeyPress!: EventEmitter<KeyboardEvent>
+
+  @Listen('click', { capture: true, target: 'document' })
+  listenOnClick(ev: UIEvent) {
+    if (this.disabled && ev.target && ev.target === this.el) {
+      ev.preventDefault()
+      ev.stopPropagation()
+    }
+  }
+
+  componentWillLoad() {
+    this.inheritedAttributes = inheritAttributes(this.el, ['aria-label', 'tabindex', 'title'])
+  }
+
+  componentDidLoad() {
+    this.didInit = true
+    if (!isNil(this.value)) {
+      this.valueChanged(this.value, undefined)
+    }
+  }
+
+  connectedCallback() {
+    this.debounceChanged()
+    attachComponentToConfig(this)
+  }
+
+  disconnectedCallback() {
+    detachComponentToConfig(this)
+  }
+
+  configChanged(state: BalConfigState): void {
+    this.language = state.language
+    this.region = state.region
+
+    if (!this.hasFocus && this.nativeInput) {
+      this.nativeInput.value = this.getFormattedValue()
+    }
+  }
+
+  /**
+   * Sets focus on the native `input` in `ion-input`. Use this method instead of the global
+   * `input.focus()`.
+   */
+  @Method()
+  async setFocus() {
+    if (this.nativeInput) {
+      this.nativeInput.focus()
+    }
+  }
+
+  /**
+   * Sets blur on the native `input` in `ion-input`. Use this method instead of the global
+   * `input.blur()`.
+   * @internal
+   */
+  @Method()
+  async setBlur() {
+    if (this.nativeInput) {
+      this.nativeInput.blur()
+    }
+  }
+
+  /**
+   * Returns the native `<input>` element used under the hood.
+   */
+  @Method()
+  getInputElement(): Promise<HTMLInputElement> {
+    return Promise.resolve(this.nativeInput!)
+  }
+
+  private getAllowedKeys() {
+    return [...NUMBER_KEYS, ...ACTION_KEYS, getDecimalSeparator()]
+  }
+
+  private getRawValue(): string {
+    return typeof this.value === 'number' ? this.value.toString() : (this.value || '').toString()
```

Comment:
isn't his always printing to string, does it metter which type is this value? Maybe just this is enough `(this.value || '').toString()` ?

Reply:
we need to check that because of 0